In [0]:
import pandas as pd

In [0]:
from google.colab import files
uploaded = files.upload()

Saving IRIS.csv to IRIS.csv


In [0]:
iris = pd.read_csv('IRIS.csv')

In [0]:
iris.head()

sepal_length  sepal_width  petal_length  petal_width      species
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa
3           4.6          3.1           1.5          0.2  Iris-setosa
4           5.0          3.6           1.4          0.2  Iris-setosa

**Question 1: How many of each flower are there?**

In [0]:
iris['species'].value_counts()

Iris-virginica     50
Iris-versicolor    50
Iris-setosa        50
Name: species, dtype: int64

**The data set is evenly split between the three different species of iris (50 of each)**

**Question 2: What is the average sepal area?**

In [0]:
round(iris['sepal_length'].mean()*iris['sepal_width'].mean(), 2)

17.85

**The average pedal length is 17.85 cm^2**

**Question 3: What is the average petal area?**

In [0]:
round(iris['petal_length'].mean()*iris['petal_width'].mean(), 2)

4.51

**The average pedal length is 4.51 cm^2**

**Question 3: Which species has the largest sepal length?**

In [0]:
iris['sepal_length'].groupby(iris['species']).mean()

species
Iris-setosa        5.006
Iris-versicolor    5.936
Iris-virginica     6.588
Name: sepal_length, dtype: float64

**Looks like virginica, following by versicolor, and finally setosa**

****Question 4: Which species has the smallest petal to sepal length ratio?****

In [0]:
(iris['petal_length'].groupby(iris['species']).mean())/(iris['sepal_length'].groupby(iris['species']).mean())

species
Iris-setosa        0.292449
Iris-versicolor    0.717655
Iris-virginica     0.842744
dtype: float64

**Setosa by far**